src/allmydata/test/no_network.py

```diff
@@ -300,6 +307,20 @@ def __init__(self, basedir, num_clients, num_servers,
             d = self.make_client(i)
             d.addCallback(lambda c: self.clients.append(c))
 
+            def _bad(f):
+                self._setup_errors.append(f)
+            d.addErrback(_bad)
+
+    def _check_clients(self):
+        """
+        The anti-pattern of doing async work in __init__ means we need to
+        check if that work completed successfully. This method either
+        returns nothing or raises an exception in case __init__ failed
+        to complete properly
+        """
+        if self._setup_errors:
+            raise self._setup_errors[0].value
+
```

Comment:
Seems there was one already: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2951

Reply:
I feel like aborting this PR because of the above is .. overkill? What I put in here was I guess just pragmatic: failing silently on e.g. a syntax error is bad, so this is the minimum to spit out something if grid-setup fails.

Reply:
(but, i could take it out and do a separate PR .. that should be ultimately fixed by 2951 above?)

Reply:
> I feel like aborting this PR because of the above is .. overkill?

I agree.  Let's not do that.